examples/runRieszWaveletAnalysis.cpp

```diff
@@ -0,0 +1,396 @@
+
+  using HeldForwardWaveletType = itk::WaveletFrequencyForward< ComplexImageType, ComplexImageType, HeldWaveletFilterBankType >;
+  auto heldForwardWavelet = HeldForwardWaveletType::New();
+
+  using HeldInverseWaveletType = itk::WaveletFrequencyInverse< ComplexImageType, ComplexImageType, HeldWaveletFilterBankType >;
+  auto heldInverseWavelet = HeldInverseWaveletType::New();
+
+  using VowForwardWaveletType = itk::WaveletFrequencyForward< ComplexImageType, ComplexImageType, VowWaveletFilterBankType >;
+  auto vowForwardWavelet = VowForwardWaveletType::New();
+
+  using VowInverseWaveletType = itk::WaveletFrequencyInverse< ComplexImageType, ComplexImageType, VowWaveletFilterBankType >;
+  auto vowInverseWavelet = VowInverseWaveletType::New();
+
+  using SimoncelliForwardWaveletType = itk::WaveletFrequencyForward< ComplexImageType, ComplexImageType, SimoncelliWaveletFilterBankType >;
+  auto simoncelliForwardWavelet = SimoncelliForwardWaveletType::New();
+
+  using SimoncelliInverseWaveletType = itk::WaveletFrequencyInverse< ComplexImageType, ComplexImageType, SimoncelliWaveletFilterBankType >;
+  auto simoncelliInverseWavelet = SimoncelliInverseWaveletType::New();
+
+  using ShannonForwardWaveletType = itk::WaveletFrequencyForward< ComplexImageType, ComplexImageType, ShannonWaveletFilterBankType >;
+  auto shannonForwardWavelet = ShannonForwardWaveletType::New();
+
+  using ShannonInverseWaveletType = itk::WaveletFrequencyInverse< ComplexImageType, ComplexImageType, ShannonWaveletFilterBankType >;
+  auto shannonInverseWavelet = ShannonInverseWaveletType::New();
+
+  if ( dimension == 2 )
+    {
+    if ( waveletFunction == "Held" )
+      {
+      return runRieszWaveletPhaseAnalysis< 2, HeldWavelet >( inputImage,
+        outputImage,
+        inputLevels,
+        inputBands,
+        applySoftThreshold,
+        thresholdNumOfSigmas);
+      }
+    else if ( waveletFunction == "Vow" )
+      {
+      return runRieszWaveletPhaseAnalysis< 2, VowWavelet >( inputImage,
+        outputImage,
+        inputLevels,
+        inputBands,
+        applySoftThreshold,
+        thresholdNumOfSigmas);
+      }
+    else if ( waveletFunction == "Simoncelli" )
+      {
+      return runRieszWaveletPhaseAnalysis< 2, SimoncelliWavelet >( inputImage,
+        outputImage,
+        inputLevels,
+        inputBands,
+        applySoftThreshold,
+        thresholdNumOfSigmas);
+      }
+    else if ( waveletFunction == "Shannon" )
+      {
+      return runRieszWaveletPhaseAnalysis< 2, ShannonWavelet >( inputImage,
+        outputImage,
+        inputLevels,
+        inputBands,
+        applySoftThreshold,
+        thresholdNumOfSigmas);
+      }
+    else
+      {
+      std::cerr << " failed!" << std::endl;
+      std::cerr << argv[5] << " wavelet type not supported." << std::endl;
+      return EXIT_FAILURE;
+      }
+    }
+  else if ( dimension == 3 )
```

Comment:
You can reduce code duplication if you make one function handle pixel type, e.g. like [this](https://github.com/InsightSoftwareConsortium/ITKMontage/blob/458ce49c2b8e4317f44c0ff775acad0a34870eaf/examples/Montage2D.cxx#L143-L172), and another `switch` handles dimension (or component type like [here](https://github.com/InsightSoftwareConsortium/ITKMontage/blob/458ce49c2b8e4317f44c0ff775acad0a34870eaf/examples/Montage2D.cxx#L216-L227))

Reply:
Cool, that's handy indeed. Thanks.

Reply:
a new template function with the "switch" between wavelet types has been added. This avoid the duplication. THanks!